aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -557,8 +568,11 @@ static inline void shapeCheck3d(
   int dimd = 1;
   int dimslices = 0;
 
-  TORCH_CHECK(input.numel() > 0 && (input.dim() == 4 || input.dim() == 5),
-      "non-empty 4D or 5D (batch mode) tensor expected for input, but got: ", input);
+  // allow batch size of 0-dim.
+  TORCH_CHECK(
+      (input.dim() == 4 && input.size(1) != 0 && input.size(2) != 0 && input.size(3) != 0) ||
```

Comment:
nit: maybe use std::accumulate with std::multiplies here and in similar places, so that you don't have to specialize for dimensions.

Reply:
I think `all_of` makes the code less terse since it only accepts a starting and ending iterator. However I think replacing the long conditions with a single bool variable can be a good idea.